lib/ymlr/encode.ex

```diff
@@ -106,6 +106,9 @@ defmodule Ymlr.Encode do
   @single_quote_when_exact [
     "",
     "~",
+    "?",
+    "-",
```

Comment:
One of your proposed test cases to cover _all_ indicators on their own failed. Adding this to include `-` among those that should be single quoted when exact caused that test to pass.

Reply:
I see. Thanks for the fix!